src/wallet/wallettool.cpp

```diff
@@ -21,16 +21,19 @@ static void WalletToolReleaseWallet(CWallet* wallet)
     delete wallet;
 }
 
-static void WalletCreate(CWallet* wallet_instance)
+static void WalletCreate(CWallet* wallet_instance, uint64_t wallet_creation_flags)
 {
     LOCK(wallet_instance->cs_wallet);
 
     wallet_instance->SetMinVersion(FEATURE_HD_SPLIT);
+    wallet_instance->AddWalletFlags(wallet_creation_flags);
 
-    // generate a new HD seed
-    auto spk_man = wallet_instance->GetOrCreateLegacyScriptPubKeyMan();
-    CPubKey seed = spk_man->GenerateNewSeed();
-    spk_man->SetHDSeed(seed);
+    if (!wallet_instance->IsWalletFlagSet(WALLET_FLAG_DESCRIPTORS)) {
+        auto spk_man = wallet_instance->GetOrCreateLegacyScriptPubKeyMan();
+        spk_man->SetupGeneration(false);
+    } else {
+        wallet_instance->SetupDescriptorScriptPubKeyMans();
+    }
```

Comment:
For a followup, we should de-duplicate this logic with the setup in `CWallet::Create`.

Reply:
That is what I was thinking too. The main challenge is that `CWallet::Create` requires a `chain` object which is used in quite a few places. Other than that it does what we need. I have a few ideas how to deal with it:

a) create dummy chain object that would return only zero/null values
b) make `chain` argument optional and check every time before using it
c) chip away a piece of `CWallet::Create` that could be used without chain into a separate function

@achow101  What do you think is the best option in your opinion?

Reply:
re: https://github.com/bitcoin/bitcoin/pull/20365#discussion_r522771512

Offhand (c) sounds great, (b) would be also be good, and it would be nice to avoid (a). The CWallet class is huge and to the extent you can pull out more functionality into standalone functions that's a good thing. Especially any functionality that does not require access to a chain state and a running node is a good candidate for being pulled out of CWallet.

Commit be164f9cf89b123f03b926aa980996919924ee64 from #15719 is kind of related (best viewed with `git log -p -n1 -w --color-moved=dimmed_zebra be164f9cf89b123f03b926aa980996919924ee64` since github display doesn't detect moved code). It moves some online functionality out of CWallet::Create into a new CWallet::AttachChain method.

Reply:
Agree with @ryanofsky. The best option I think is to break apart `CWallet::Create` and refactor pieces of it into standalone functions.

Reply:
@ryanofsky @achow101 thanks for the advice and the reference. I'll start working on the refactoring in separate PR as I think it's better not to mix things up.

I've extended functional tests to cover for the new parameter. Any comments on this PR?